fix(flask): stop reading deprecated flask.__version__ (#4422)
open-telemetry/opentelemetry-python-contrib: CHANGELOG.md

- Bump `pylint` to `4.0.5`
  ([#4244](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/4244))

### Fixed

> - `opentelemetry-instrumentation-flask`: Stop reading the deprecated `flask.__version__` attribute; resolve the Flask version via `importlib.metadata` only so import no longer emits a `DeprecationWarning` under Flask 3.1+

Reviewer: ```suggestion
- `opentelemetry-instrumentation-flask`: Stop reading the deprecated (from 3.1) `flask.__version__` attribute; resolve the Flask version via `importlib.metadata`
```
